mne/dipole.py

```diff
@@ -342,43 +342,54 @@ def _fit_Q(fwd_data, whitener, proj_op, B, B2, B_orig, rd):
     return Q, gof, B_residual
 
 
-def _fit_dipoles(data, times, rrs, guess_fwd_svd, fwd_data, whitener,
-                 proj_op, n_jobs):
+def _fit_dipoles(min_dist_to_inner_skull, data, times, rrs, guess_fwd_svd,
+                 fwd_data, whitener, proj_op, n_jobs):
     """Fit a single dipole to the given whitened, projected data"""
     from scipy.optimize import fmin_cobyla
     parallel, p_fun, _ = parallel_func(_fit_dipole, n_jobs)
     # parallel over time points
-    res = parallel(p_fun(B, t, rrs, guess_fwd_svd, fwd_data, whitener, proj_op,
-                         fmin_cobyla)
+    res = parallel(p_fun(min_dist_to_inner_skull, B, t, rrs, guess_fwd_svd,
+                         fwd_data, whitener, proj_op, fmin_cobyla)
                    for B, t in zip(data.T, times))
     pos = np.array([r[0] for r in res])
     amp = np.array([r[1] for r in res])
     ori = np.array([r[2] for r in res])
     gof = np.array([r[3] for r in res]) * 100  # convert to percentage
     residual = np.array([r[4] for r in res]).T
+
     return pos, amp, ori, gof, residual
 
 
-def _fit_dipole(B_orig, t, rrs, guess_fwd_svd, fwd_data, whitener, proj_op,
+def _fit_dipole(min_dist_to_inner_skull, B_orig, t, rrs,
+                guess_fwd_svd, fwd_data, whitener, proj_op,
                 fmin_cobyla):
     """Fit a single bit of data"""
-    logger.info('---- Fitting : %7.1f ms' % (1000 * t,))
     B = np.dot(whitener, B_orig)
+    assert min_dist_to_inner_skull >= 0.
 
+    surf = None
     # make constraint function to keep the solver within the inner skull
     if isinstance(fwd_data['inner_skull'], dict):  # bem
         surf = fwd_data['inner_skull']
 
         def constraint(rd):
```

Comment:
IIRC the nesting is necessary here because of use of function local variables. We could use partial but that becomes less clear I think.